hazelcast/src/main/java/com/hazelcast/datalink/HazelcastDataLink.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.clientside.HazelcastClientProxy;
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.util.StringUtil;
+import com.hazelcast.map.IMap;
+import com.hazelcast.spi.annotation.Beta;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.jet.impl.util.ImdgUtil.asClientConfig;
+import static com.hazelcast.jet.impl.util.ImdgUtil.asClientConfigFromYaml;
+
+/**
+ * Creates a HazelcastInstance that is shared to connect to a remote cluster.
+ * <p>
+ * Set the client XML as property using {@link HazelcastDataLink#CLIENT_XML} as property key or
+ * Set the client YAML as property using {@link HazelcastDataLink#CLIENT_YML} as property key
+ *
+ * @since 5.3
+ */
+@Beta
+public class HazelcastDataLink extends DataLinkBase implements DataLink {
```

Comment:
If the class extends DataLinkBase it already implements DataLink

Reply:
Yes, I find this clearer - see e.g. collections in `java.util` do the same.